docs/src/pages/Conformance.md

```diff
@@ -295,6 +301,19 @@ Elements of type Reference may contain a versioned reference, such as `Patient/1
     * This is because the IBM FHIR Server will only return the current version of `match` resources in search results. A reference to a non-current version of the resource is not considered to have met the search criteria, thus the resource containing the reference is not considered a valid `include` resource.
     * Example: A `Condition` resource contains a reference of `Patient/123/_history/2` in its `subject` element, and the current version of the `Patient/123` resource is `2`, and a search of `Patient?_revinclude=Condition:subject` is performed. The `Condition` resource will be returned as an `include` resource since the version of the `Patient` resource specified in the `subject` element is the current version of the `Patient` resource. If the current version of the `Patient/123` resource is `3`, then the `Condition` resource will not be returned as an `include` resource in the search results.
 
+**Search on Elements of Type Canonical:**
+
+Canonical searches search against the canonical URL of a resource. This is for resource types that have a defined element `url` which is the URL that always identifies the resource across all contexts of use. The list of resource types that have a canonical URL and are allowed to be the target of a reference to a canonical URL are documented in the [FHIR specification](http://build.fhir.org/references.html#canonical).
+
+Canonical searches are processed following these rules:
+* The search parameter value is a reference to a definitional FHIR resource (of a type from the list documented in the FHIR specification), where the reference is to the resource’s indexed `url` element value.
+    * The reference can optionally contain a version, delimited by the `|` character (i.e. `http://example.org/fhir/ValueSet/123|1.0.0`). The version will be automatically detected and interpreted as a reference to a resource’s business version - the indexed `version` element value (not to be confused with a resource’s `meta.versionId` element).
+* Searches which do not specify a version will match against all resources whose indexed `url` value matches the search parameter value, regardless if the resource has a `version` value specified.
+* Searches which do specify a version will match only against those resources whose indexed `url` and indexed `version` values match the `url` and `version` components of the search parameter value.
+* Canonical search parameters may be used when performing chained, reverse chained (`_has`), `_include`, or `_revinclude` searches. As described above, versioned canonical references act against the business version of a resource, not it’s versionId. The special rules that apply to chaining and include references specifying versions of the form `/_history/xx` do not apply to versioned canonical references. Versions of the form `/_history/xx` are not supported with canonical references.
```

Comment:
Is "canonical search parameters" defined somewhere?  I think it means search parameters of type `reference` which target an element of type canonical.  But I'm don't think we know which ones those are when we process a search query, so maybe its just reference search params that have a value which looks like a canonical url?

Reply:
OK, I think I found my answer in the code.  Specifically SearchUtil.isCanonicalSearchParm is doing some work (understatement) to determine if the search parameter can target elements of type canonical or not.

Reply:
I updated this section a bit. Hopefully a little clearer, but you're right - 'canonical search parameter' is determined by looking at the FHIRPath definition

Reply:
looks good